org.osgi.service.feature/src/org/osgi/service/feature/ID.java

```diff
@@ -34,125 +33,58 @@
-	 * syntax:
-	 * <p>
-	 * {@code group-id ':' artifact-id [ ':' [type] [ ':' classifier ] ] ':' version}
-	 *
-	 * @param mavenID
-	 * @return The ID
+public interface ID {
+	/**
+	 * ID type for use with Features.
 	 */
-    public static ID fromMavenID(String mavenID) {
-        String[] parts = mavenID.split(":");
-
-        if (parts.length < 3 || parts.length > 5)
-            throw new IllegalArgumentException("Not a valid maven ID" + mavenID);
-
-        String gid = parts[0];
-        String aid = parts[1];
-        String ver = parts[2];
-        String t = parts.length > 3 ? parts[3] : null;
-        String c = parts.length > 4 ? parts[4] : null;
-
-        return new ID(gid, aid, ver, t, c);
-    }
-
-    /**
-     * Construct an Artifact ID
-     * @param groupId The group ID.
-     * @param artifactId The artifact ID.
-     * @param version The version.
-     */
-    public ID(String groupId, String artifactId, String version) {
-        this(groupId, artifactId, version, null, null);
-    }
+	static final String FEATURE_ID_TYPE = "osgifeature";
 
-    /**
-     * Construct an Artifact ID
-     * @param groupId The group ID.
-     * @param artifactId The artifact ID.
-     * @param version The version.
-     * @param type The type identifier.
-     * @param classifier The classifier.
-     */
-    public ID(String groupId, String artifactId, String version, String type, String classifier) {
-        this.groupId = groupId;
-        this.artifactId = artifactId;
-        this.version = version;
-        this.type = type;
-        this.classifier = classifier;
-    }
-
-    /**
-     * Get the group ID.
-     * @return The group ID.
-     */
-    public String getGroupId() {
-        return groupId;
-    }
-
-    /**
-     * Get the artifact ID.
-     * @return The artifact ID.
-     */
-    public String getArtifactId() {
-        return artifactId;
-    }
-
-    /**
-     * Get the version.
-     * @return The version.
-     */
-    public String getVersion() {
-        return version;
-    }
+	/**
+	 * Get the group ID.
+	 * 
+	 * @return The group ID.
+	 */
+	String getGroupId();
 
-    /**
-     * Get the type identifier.
-     * @return The type identifier.
-     */
-    public Optional<String> getType() {
-        return Optional.ofNullable(type);
-    }
+	/**
+	 * Get the artifact ID.
+	 * 
+	 * @return The artifact ID.
+	 */
+	String getArtifactId();
 
-    /**
-     * Get the classifier.
-     * @return The classifier.
-     */
-    public Optional<String> getClassifier() {
-        return Optional.ofNullable(classifier);
-    }
+	/**
+	 * Get the version.
+	 * 
+	 * @return The version.
+	 */
+	String getVersion();
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(artifactId, classifier, groupId, type, version);
-    }
+	/**
+	 * Get the type identifier.
+	 * 
+	 * @return The type identifier.
+	 */
+	Optional<String> getType();
```

Comment:
I am now thinking this should not be optional but instead be just String where the default value is `jar`

Reply:
No, if the type is not specified there is no type. In some contexts 'jar' is then used instead, but from an ID point of view, if it's not specified then there is no value.